tests/e2e/taqueria-cli-persistent-state.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { exec as exec1 } from 'child_process';
+import fsPromises from 'fs/promises';
+import path from 'path';
+import utils from 'util';
+import { generateTestProject, sleep } from './utils/utils';
+const exec = utils.promisify(exec1);
+
+const taqueriaProjectPath = 'scrap/auto-test-persistent-state';
+
+describe('Test CLI Persistent State', () => {
+	beforeEach(() => fsPromises.rm(taqueriaProjectPath, { force: true, recursive: true }));
+
+	test('Ensure that the state file named using the default config.json and no env set using the CLI', async () => {
+		await generateTestProject(taqueriaProjectPath, ['ligo']);
+		await exec(`cp e2e/data/hello-tacos.mligo ${taqueriaProjectPath}/contracts`);
+		await exec(`taq compile hello-tacos.mligo`, { cwd: taqueriaProjectPath });
+		expect(fsPromises.stat(path.join(taqueriaProjectPath, '.taq', 'development-state.json'))).resolves.not.toBeFalsy();
+	});
+
+	test('Ensure that state file is named after a valid environment when there is no default in the config', async () => {
+		await generateTestProject(taqueriaProjectPath, ['ligo']);
+		await exec(`cp e2e/data/config-without-default-environment.json ${taqueriaProjectPath}/contracts`);
+		await exec(`cp e2e/data/hello-tacos.mligo ${taqueriaProjectPath}/contracts`);
+		await exec(`taq compile hello-tacos.mligo`, { cwd: taqueriaProjectPath });
+		expect(fsPromises.stat(path.join(taqueriaProjectPath, '.taq', 'development-state.json'))).resolves.not.toBeFalsy();
+	});
+
+	test('Ensure that state file is named after testing environment when the default in the config is set to testing', async () => {
+		await generateTestProject(taqueriaProjectPath, ['ligo']);
+		await exec(`cp e2e/data/config-without-default-environment-testing.json ${taqueriaProjectPath}/contracts`);
+		await exec(`cp e2e/data/hello-tacos.mligo ${taqueriaProjectPath}/contracts`);
+		await exec(`taq compile hello-tacos.mligo`, { cwd: taqueriaProjectPath });
+		expect(fsPromises.stat(path.join(taqueriaProjectPath, '.taq', 'testing-state.json'))).resolves.not.toBeFalsy();
+	});
+
+	test('Ensure that state file is named after testing environment when the env is specified using the CLI', async () => {
+		await generateTestProject(taqueriaProjectPath, ['ligo']);
+		await exec(`cp e2e/data/hello-tacos.mligo ${taqueriaProjectPath}/contracts`);
+		await exec(`taq compile -e testing hello-tacos.mligo`, { cwd: taqueriaProjectPath });
+		await sleep(2000); // for some reason this is needed to give time for the testing-state.json file to be created
```

Comment:
can't you do an await that checks for testing-state.json having been created?

Reply:
@michaelkernaghan I've gone and ahead and added a `waitFor()` utility that will continuously poll till a promise resolves. Mind giving things another review please?

Reply:
ok.